proto/rig/model/pipeline.proto

```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
+
+package model;
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+
+message Pipeline {
+  // Custom name for the pipeline.
+  string name = 1;
+  repeated Phase phases = 2;
```

Comment:
What is the logic behind phase transitions? If I have two phases, one for `staging` and one for `prod`, does it then transition from `staging` to `prod`, keeping the fixed_fields defined on the `staging` or `prod` phase?

Reply:
I was thinking the following:
It computes the changes from 'prod' to 'staging', and removes changes that affect fixed_fields in 'prod'

Reply:
Ah so the destination phase defines how to process the promotion, but the source phase defines when we transition?